### DISCUSSION TITLE
Zksync deploying takes much time | lesson 15

### DISCUSSION THREAD

**Discussion by jnow1233** [NethermindEth/zksync-remix-plugin#154] [General]
Hi,
I'm trying to deploy a contract to zksync but it takes much time without any action, is it normal ?
how long it should takes to deploy ?
![image](https://github.com/user-attachments/assets/5c646550-e8bb-4acb-aae6-f0b511c62656)

**Comment by varex83** [comment]
Hello @jnow1233 ! 

Are you trying to deploy to the devnet or using the wallet connection?

**Reply by jnow1233** [reply]
Hi thanks for your reply.
using wallet connection similar to the video, it works correctly with solidity compiler and sepolia testnet everything ok, the problem is with deploying in zksync

**Reply by varex83** [reply]
Okay, got it, sometimes the interaction with wallet could make some troubles since our plugin is connected via iframe, which is not very standard for web3 integration, we will add `InjectedProvider` support in the following release, so it will reduce such troubles
